### PR TITLE
(TESTING) Alpine 3.9.0

### LINKED ISSUE #133: docker-compose not starting on first startup
## Comments

Have had it happen twice already. But after I issue a **/etc/init.d/docker-compose start** the problem goes away.

Any thoughts?

## Issue

in **/var/log/rc.log**

```

Pulling btcbox (lncm/bitcoind:0.17.1-alpine-arm-wallet)...                       0.17.1-alpine-arm-wallet: Pulling from lncm/bitcoind                                                                                                              5b678b67777f: Pulling fs layer                                                   d9f0b2b885d9: Pulling fs layer                                                   32449c89e9bd: Pulling fs layer                                                   b495499d71c4: Pulling fs layer                                                   2eaca0c61e44: Pulling fs layer                                                   590deaa36de9: Pulling fs layer2eaca0c61e44: Waitingb495499d71c4: Waiting590deaa3 6de9: WaitingERROR: error pulling image configuration: Get https://production.cl oudflare.docker.com/registry-v2/docker/registry/v2/blobs/sha256/e8/e8ddecd9b1649 983c39aec154d5a1b348cc83a16aee884b976be43267746fa29/data?verify=1548846106-Z9%2F Th%2BqawCvk81C1rcJcaTNP%2Bu0%3D: x509: certificate has expired or is not yet val id                                                                                [ !! ]                                                                           * ERROR: docker-compose failed to start
--



```

### LINKED ISSUE #140: Update Alpine to the latest and investigate use of postmarketOS for orchestration
Also verify if `make_modloop.sh` is no longer necessary, and if that's the case, remove it.

### PR DESCRIPTION
**Pull request by AnotherDroog**
Closes #140 Update alpine to 3.9.0

* Alpine release 3.9.0
* Updates to most packages
* Updated cache.tar.gz (docker, avahi, etc)
* `make_cache.sh` creates __cache.tar.gz__ under alpine
* kernel update to 4.19.18

### Outstanding issues:
- [ ] WiFi / `wlan0`  on __3B+__ is broken (again)
_Since alpine 3.9.0 `modloop` files are signed, therefore patching is difficult
The issue can be worked around by doing initial installation via LAN cable_

### Fixes
- [x] Fixes #133  tls certificate errors during container installation

### CODE REVIEW

**Comment by nolim1t** [comment]
I would hold off if wlan is broken

**Comment by meeDamian** [comment]
> I would hold off if wlan is broken

@AnotherDroog Is there any other reason to upgrade now? If not, let's hold off.

**Comment by AnotherDroog** [comment]
Box 0.4.1 image:
https://ipfs.io/ipfs/QmX83AUBMAZ39wXJbNvUkSdEhHkTV9WeLtYyrTobLEoZYg

**Comment by AnotherDroog** [comment]
new cache.tar.gz:
https://ipfs.io/ipfs/Qme2nfpmvnseEK8F6cTXwBxv2QmyqN3oTqtzRm1Gh7NyLz

File: etc/init.d/lncm-post [hunk 1]
```diff
@@ -163,6 +163,12 @@ start() {
         /bin/cp -f /home/lncm/nginx/mime.types /media/important/nginx/mime.types  
     fi
 
+    echo "Download index.html"
+    if [ ! -f /home/lncm/public_html/index.html]; then
+      cd /home/lncm/public_html
+      wget https://raw.githubusercontent.com/lncm/invoicer-ui/master/dist/index.html
```
Comment (nolim1t) [inline etc/init.d/lncm-post]: We need to change invoicer-ui first to not use "Food4Thought" in the name but have it standard as "LNCM"

File: home/lncm/compose/docker-compose.yml [hunk 2]
```diff
@@ -1,79 +1,81 @@
...
-        ports:
-            - "80:80"
-        depends_on:
-            - invoicerbox
-        restart: on-failure
-        networks:
-            localnet:
-                ipv4_address: 172.16.88.250
+  bitcoind:
+    image: lncm/bitcoind:0.17.1
```
Comment (nolim1t) [inline home/lncm/compose/docker-compose.yml]: does this tag exist?

**Comment by AnotherDroog** [comment]
There’s an issue with bitcoind IP in docker compose file. Needs to be changed to .2

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by AnotherDroog** [comment]
Good news is that upstream alpine WiFi bug is confirmed fixed now

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by AnotherDroog** [comment]
>There’s an issue with bitcoind IP in docker compose file. Needs to be changed to .2 ...

@AnotherDroog fixed

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by nolim1t** [comment]
Please get #161 into this branch. It's the changes that we made to get the box working at Cube

**Comment by AnotherDroog** [comment]
I’m going to hold off merging until 3.9.1 is out, with the WiFi fix included. 

Let’s backport everything here that’s not related to 3.9.1 into master

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by meeDamian** [comment]
Perfect 😄

**Comment by AnotherDroog** [comment]
There’s a `3.8.2-backports` branch now, which will be merged with master once we’re sure it’s stable

**Comment by nolim1t** [comment]
We should consider getting the stuff from #185 into this branch too. It makes the box a tor node by default (but can be switched off if needed)

**Comment by nolim1t** [comment]
re that conflict with master, those files are now symlinks to the tor version of the config now.

**Comment by nolim1t** [comment]
We need to change ```home/lncm/compose/docker-compose.yml``` and ```home/lncm/lnd/lnd.conf``` to match master.

But I'll probably also change docker-compose config a bit more to finally remove the bridge networking from docker completely

File: home/lncm/compose/docker-compose.yml [hunk 3]
```diff
@@ -1,79 +1,81 @@
-version: '2.1'
+version: '3.7'
```
[the remark this answers is not in the document]
Comment (AnotherDroog) [reply home/lncm/compose/docker-compose.yml]: After looking into the matter, it turns out docker-compose versions are linked to certain versions of Docker. It's not a good idea to change this randomly!

**Comment by AnotherDroog** [comment]
Going to close and re-open when Alpine ships the next release in ~2 weeks

**Pull request closed without merging**